libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -796,6 +798,19 @@ export class Simulator {
     // allocate a handle for the resource so others can find it
     const handle = this._handles.allocate(path, resourceObject);
 
+    // if the resource is a policy, add it to the policy registry
+    if (resourceConfig.type === POLICY_FQN) {
```

Comment:
Do we still really need this resource?

Reply:
> Do we still really need this resource?

I mentioned how sim.Policy behaves differently in the PR description:

> In general, `sim.Policy` is more flexible in the current implementation since using it doesn't automatically introduce a dependency on the resource(s) permissions are granted to. For example, a `Topic` can have external policy specifying that it can invoke a `Function`, without necessarily introducing a dependency between the `Topic` and `Function`.

Reply:
I see. It makes sense